src/main/java/frc/robot/subsystems/shooter/ShooterIOTalonFX.java

```diff
@@ -0,0 +1,373 @@
+      angleMotorConfig.kP = rotationMotorKP.get();
+      angleMotorConfig.kI = rotationMotorKI.get();
+      angleMotorConfig.kD = rotationMotorKD.get();
+      angleMotorConfig.kS = rotationMotorKS.get();
+      angleMotorConfig.kG = rotationMotorKG.get();
+      angleMotorConfig.kA = rotationMotorKA.get();
+      angleMotorConfig.kV = rotationMotorKV.get();
+      angleMotor.getConfigurator().apply(angleMotorConfig);
+
+      MotionMagicConfigs rotationMotionMagicConfig = new MotionMagicConfigs();
+      rotationMotionMagicConfig.MotionMagicExpo_kV = rotationMotorExpoKV.get();
+      rotationMotionMagicConfig.MotionMagicExpo_kA = rotationMotorExpoKA.get();
+      rotationMotionMagicConfig.MotionMagicCruiseVelocity =
+          ShooterConstants.MOTION_MAGIC_CRUISE_VELOCITY;
+      rotationMotionMagicConfig.MotionMagicAcceleration =
+          ShooterConstants.MOTION_MAGIC_CRUISE_VELOCITY * 2;
+      angleMotor.getConfigurator().apply(rotationMotionMagicConfig);
+    }
+  }
+
+  @Override
+  public void setShooterWheelTopVelocity(double rps) {
+    shootMotorTop.setControl(shootMotorTopVelocityRequest.withVelocity(rps));
+  }
+
+  @Override
+  public void setShooterWheelBottomVelocity(double rps) {
+    shootMotorBottom.setControl(shootMotorBottomVelocityRequest.withVelocity(rps));
+  }
+
+  @Override
+  public void setAngle(double angle) {
+    angleMotor.setControl(angleMotorPositionRequest.withPosition(degreesToRotations(angle)));
+  }
+
+  private double degreesToRotations(double degrees) {
+    return degrees / 360;
+  }
+
+  private double rotationsToDegrees(StatusSignal<Double> statusSignal) {
+    return statusSignal.getValueAsDouble() * 360;
+  }
+
+  private void configShootMotor(TalonFX shootMotor, boolean isInverted) {
+
+    TalonFXConfiguration shootMotorsConfig = new TalonFXConfiguration();
+    CurrentLimitsConfigs shootMotorsCurrentLimits = new CurrentLimitsConfigs();
+
+    shootMotorsCurrentLimits.SupplyCurrentLimit =
+        ShooterConstants.SHOOT_MOTORS_CONTINUOUS_CURRENT_LIMIT;
+    shootMotorsCurrentLimits.SupplyCurrentThreshold =
+        ShooterConstants.SHOOT_MOTORS_PEAK_CURRENT_LIMIT;
+    shootMotorsCurrentLimits.SupplyTimeThreshold =
+        ShooterConstants.SHOOT_MOTORS_PEAK_CURRENT_DURATION;
+    shootMotorsCurrentLimits.SupplyCurrentLimitEnable = true;
+
+    shootMotorsConfig.CurrentLimits = shootMotorsCurrentLimits;
+
+    shootMotorsConfig.MotorOutput.NeutralMode = NeutralModeValue.Coast;
+    shootMotorsConfig.Slot0.kP = shootMotorsKP.get();
+    shootMotorsConfig.Slot0.kI = shootMotorsKI.get();
+    shootMotorsConfig.Slot0.kD = shootMotorsKD.get();
+    shootMotorsConfig.Slot0.kS = shootMotorsKS.get();
+
+    shootMotorsConfig.Feedback.SensorToMechanismRatio = ShooterConstants.SHOOT_MOTORS_GEAR_RATIO;
+
+    shootMotorsConfig.MotorOutput.Inverted =
+        isInverted ? InvertedValue.Clockwise_Positive : InvertedValue.CounterClockwise_Positive;
+
+    shootMotor.getConfigurator().apply(shootMotorsConfig);
```

Comment:
loop to ensure configuration is applied and set an alert if it isn't (look at Intake for an example):

```
StatusCode status = StatusCode.StatusCodeNotInitialized;
    for (int i = 0; i < 5; ++i) {
      status = kickerMotor.getConfigurator().apply(kickerConfig);
      if (status.isOK()) break;
    }
    if (!status.isOK()) {
      configAlert.set(true);
      configAlert.setText(status.toString());
    }
```